core/scripts/liquidity-mining/CalculateBalancerLPProviders.js

```diff
@@ -0,0 +1,155 @@
+// This script calculates $UMA liquidity mining rewards for Balancer Pools. This is done with the following process:
+// -> Define the starting and ending blockheight of each week. Both are chosen as the block with the closest timestamp
+// to a fixed weekly time(e.g.2020-06-06 00:00pm UTC).
+// -> Define snapshot blocks (eg every 64 blocks ~15min). For each snapshot block, calculate the proportional liquidity
+// provided by for each liquidity provider to the single whitelisted pool.
+// -> For each snapshot block, calculate the $UMA rewards to be received by each liquidity provider based on the target weekly distribution.
+
+// Example usage from core: node ./scripts/liquidity-mining/CalculateBalancerLPProviders.js --poolAddress="0x0099447ef539718bba3c4d4d4b4491d307eedc53" --fromDate="2020-07-06" --toDate="2020-07-13" --week=1
+
+const moment = require("moment");
+const cliProgress = require("cli-progress");
+const fs = require("fs");
+const path = require("path");
+const Web3 = require("web3");
+const utils = require("./utils");
+const poolAbi = require("../../build/contracts/ERC20.json");
+require("dotenv").config();
+
+const web3 = new Web3(new Web3.providers.HttpProvider(process.env.CUSTOM_NODE_URL));
+
+const { toWei, toBN, fromWei } = web3.utils;
+
+const argv = require("minimist")(process.argv.slice(), {
+  string: ["poolAddress", "fromDate", "toDate"],
+  integer: ["week"],
+  boolean: ["test"]
+});
+
+const UMA_PER_WEEK = toBN(toWei("25000"));
+const BLOCKS_PER_SNAPSHOT = 64;
+let umaPerSnapshot;
+
+(async function() {
```

Comment:
Just curious, do you think this would be clearer if we structured it like this:
```js
async function main() {
    // Logic
};

main().then(() => {
    // Do nothing maybe?
}).catch(err => {
    // Handle errors
}
```

Just to separate concern between process-level management (the .then and .catch) and logic. There may be disadvantages with doing it that way, though.

Reply:
ye, more than happy to refactor it like this :)